VideoLocker/src/main/java/org/edx/mobile/view/CourseDetailFragment.java

```diff
@@ -0,0 +1,268 @@
+        courseTextDetails.setText(CourseCardUtils.getDescription(courseDetail.org, courseDetail.number, formattedDate));
+        courseTextName.setText(courseDetail.name);
+    }
+
+    private void setCourseImage() {
+        final String headerImageUrl = environment.getConfig().getApiHostURL() + courseDetail.media.course_image.uri;
+        Glide.with(CourseDetailFragment.this)
+                .load(headerImageUrl)
+                .placeholder(R.drawable.edx_map_login)
+                .transform(new TopAnchorFillWidthTransformation(getActivity()))
+                .into(headerImageView);
+    }
+
+    private void setCourseVideoButton() {
+        if (courseDetail.media.course_video.uri == null || courseDetail.media.course_video.uri.isEmpty()) {
+            headerPlayIcon.setEnabled(false);
+        } else {
+            headerPlayIcon.setVisibility(headerPlayIcon.VISIBLE);
+        }
+    }
+
+    /**
+     * Makes a call the the course details api and sets the overview if given.
+     */
+    private void setAboutThisCourse() {
+        getCourseDetailTask = new GetCourseDetailTask(getActivity(), courseDetail.course_id) {
+            @Override
+            protected void onSuccess(CourseDetail courseDetail) throws Exception {
+                super.onSuccess(courseDetail);
+                if (courseDetail.overview != null && !courseDetail.overview.isEmpty()) {
+                    setAboutThisCourse(courseDetail.overview);
+                }
+            }
+
+            @Override
+            protected void onException(Exception e) throws RuntimeException {
+                super.onException(e);
+                showErrorMessage(e);
+            }
+        };
+        getCourseDetailTask.execute();
+    }
+
+    /**
+     * Takes a string which can include html and then renders it into the courseAbout webview.
+     *
+     * @param overview A string that can contain html tags
+     */
+    private void setAboutThisCourse(String overview) {
+        addDivider();
+        courseAbout.setVisibility(View.VISIBLE);
+        URLInterceptorWebViewClient client = new URLInterceptorWebViewClient(
+                getActivity(), courseAboutWebView);
+        client.setAllLinksAsExternal(true);
+
+        StringBuffer buff = new StringBuffer();
+        buff.append("<head>");
+        buff.append("<meta name=\"viewport\" content=\"width=device-width, initial-scale=1\">");
+        try {
+            String cssFileContent = FileUtil.loadTextFileFromAssets(getActivity(), "css/render-html-in-webview.css");
+            buff.append("<style>");
+            buff.append(cssFileContent);
+            buff.append("</style>");
+        } catch (IOException e) {
+            logger.error(e);
+        }
+        buff.append("</head>");
+        buff.append("<body>");
+        buff.append("<div class=\"header\">");
+        buff.append(overview);
+        buff.append("</div>");
+        buff.append("</body>");
```

Comment:
nit: Maybe we should share this HTML document construction with the `populateAnnouncements` method? The only difference is the content of the body and the name of the css file, which can be the two parameters.